src/hypothesis/searchstrategy/strategies.py

```diff
@@ -76,16 +193,22 @@ def accept(self):
     # The fact that this returns False does not guarantee that a valid value
```

Comment:
My understanding is that this is still correct - the new algorithm guarantees consistency but not correctness.  Right?

Reply:
Correct. The new algorithm precisely calculates the fixed point of a set of formulae, but those formulae are only a conservative approximation to whether a strategy is empty, which is an essentially undecidable problem.